src/types/table.ts

```diff
@@ -13,6 +15,16 @@ export interface TablePreferences {
   columnVisibility?: Record<string, boolean>
 }
 
+export interface TableDataEntry {
+  [key: string]: any
+  /** RouteLocationRaw or url string for row link */
+  to?: RouteLocationRaw | string
+  /** Target for row link */
+  target?: string
```

Comment:
suggestion: there only only some strings that are valid here; should we tighten the interface to only allowed values? 

(Should still discuss these “reserved” properties)

Reply:
Good call. Let's resolve the reserved properties discussion and then come back to this.